gapis/memory/pool.go

```diff
@@ -100,6 +111,49 @@ func (m *Pool) String() string {
 	return strings.Join(l, "\n")
 }
 
+// New creates and returns a new Pool and its id.
+func (m *Pools) NewPool() (id PoolID, p *Pool) {
+	id, p = m.nextPoolID, &Pool{}
+	m.pools[id] = p
+	m.nextPoolID++
+	return
+}
+
+// Get returns the Pool with the given id.
+func (m *Pools) Get(id PoolID) (*Pool, error) {
+	if p, ok := m.pools[id]; ok {
+		return p, nil
+	}
+	return nil, fmt.Errorf("Pool %v not found", id)
+}
+
+// GetOrPanic returns the Pool with the given id, or panics if it's not found.
+func (m *Pools) GetOrPanic(id PoolID) *Pool {
```

Comment:
Argh. I really hate these functions - I'm trying to remove all panics from GAPIS as we keep on hitting them, but I do see that this is no worse than what we had before.
We typically call these sorts of abominations `MustXXX`, so this would be called `MustGet`.